Shop_Unity_base/Assets/Scripts/Shop/Model/Items/PotionFactory.cs

```diff
@@ -1,23 +1,41 @@
-﻿
-namespace Model.Items
+﻿namespace Model.Items
 {
     using System;
     using System.Collections.Generic;
 
     public class PotionFactory : IItemFactory
     {
-        public enum PotionType { Health, Mana, Strength, Defence, Invisibility }
+        public enum PotionType { HealtSmall, HealthGrand, Mana, Strength, Poison, Invisibility }
```

Comment:
Now you can start to see how cumbersome this can get.

Reply:
Yup, Not sure if there's a much better way around it. Code generation? :p

Reply:
Could use an ever-expanding Dictionary. With the keys being hashes, strings, or just generated IDs, based on something dumb like A-Z order based on the item's name?

Reply:
Yeah, that sounds pretty good. I'm going to classify it as a Could-Have though, for the sake of time.